tests/rptest/tests/pandaproxy_test.py

```diff
@@ -1103,9 +1105,22 @@ def test_fetch_topic(self):
         for o in produce_result["offsets"]:
             assert o["offset"] == 0, f'error_code {o["error_code"]}'
 
+        # Partition movement may occur between the call to produce_topic
+        # and fetch_topic. This manifests as unknown_topic_or_partition.
+        # Account for this by issuing the fetch_topic request to the leader.
```

Comment:
There's nothing stopping the leader from moving between the call here and the call to consume.

The changes in the previous commit should allow mitigation and retry loop, I don't the changes to the test are required.

Reply:
My concern is that this will mask failures in the retry logic in `kafka::client`.

I think this PR can just be the second commit.

Reply:
Yeah I agree to that. I was skeptical about this code too but wanted to see what the reviews said.

Reply:
dropped.